### PR TITLE
Limit max possible threads in ThreadPool (SetMaxThreads)

### PR DESCRIPTION
**Pull request by EgorBo**
to 0x7fff like CoreCLR does:
https://github.com/dotnet/coreclr/blob/master/src/vm/win32threadpool.h#L111
https://github.com/dotnet/coreclr/blob/master/src/vm/win32threadpool.cpp#L530

I was trying to port tests from corefx to mono (for ThreadPool) and one of them tests this.
https://github.com/dotnet/corefx/blob/master/src/System.Threading.ThreadPool/tests/ThreadPoolTests.cs#L123-L124

Bugzilla: https://bugzilla.xamarin.com/show_bug.cgi?id=60027

### CODE REVIEW

File: mcs/class/corlib/Test/System.Threading/ThreadPoolTest.cs [hunk 1]
```diff
@@ -156,6 +156,23 @@ public void SetAndGetMaxThreads ()
...
+			
+			ThreadPool.SetMaxThreads(maxPossibleThreads, maxPossibleThreads);
+			ThreadPool.GetMaxThreads(out maxWt, out macCpt);
+			Assert.AreEqual(maxPossibleThreads, maxWt);
+			Assert.AreEqual(maxPossibleThreads, macCpt);
+
+			ThreadPool.SetMaxThreads(maxPossibleThreads + 1, maxPossibleThreads + 1);
+			ThreadPool.GetMaxThreads(out maxWt, out macCpt);
+			Assert.AreEqual(maxPossibleThreads, maxWt);
+			Assert.AreEqual(maxPossibleThreads, macCpt);
```
Comment (akoeplinger) [inline mcs/class/corlib/Test/System.Threading/ThreadPoolTest.cs]: code style: space before opening `(`

File: mono/metadata/threadpool.c [hunk 2]
```diff
@@ -48,6 +48,9 @@
 #include <mono/utils/refcount.h>
 #include <mono/utils/mono-os-wait.h>
 
+// consistency with coreclr https://github.com/dotnet/coreclr/blob/master/src/vm/win32threadpool.h#L111
```
Comment (akoeplinger) [inline mono/metadata/threadpool.c]: please don't use a link to `master` as that'll quickly get outdated. always use a permalink like https://github.com/dotnet/coreclr/blob/643b09f966e68e06d5f0930755985a01a2a2b096/src/vm/win32threadpool.h#L111

File: mcs/class/corlib/Test/System.Threading/ThreadPoolTest.cs (hunk 1 above)
Comment (EgorBo) [reply mcs/class/corlib/Test/System.Threading/ThreadPoolTest.cs]: Oh I always forget :( Fixed.

**Comment by marek-safar** [comment]
This fails to compile due in C

**Pull request merged**